packages/flutter/test/material/range_slider_test.dart

```diff
@@ -0,0 +1,1214 @@
+                        values = newValues;
+                      });
+                    },
+                  ),
+                ),
+              ),
+            );
+          },
+        ),
+      ),
+    );
+
+    // Get the bounds of the track by shifting by the the overlay radius.
+    final Offset topLeft = tester.getTopLeft(find.byType(RangeSlider)).translate(24, 0);
+    final Offset bottomRight = tester.getBottomRight(find.byType(RangeSlider)).translate(-24, 0);
+
+    // Drag the end thumb to the max.
+    final Offset leftTarget = topLeft + (bottomRight - topLeft) * 0.3;
+    await tester.dragFrom(leftTarget, topLeft + (bottomRight - topLeft) * -0.4);
+    expect(values.end, equals(1));
+
+    // Drag the start thumb to the min.
+    await tester.pumpAndSettle();
+    final Offset rightTarget = topLeft + (bottomRight - topLeft) * 0.7;
+    await tester.dragFrom(rightTarget, topLeft + (bottomRight - topLeft) * 0.4);
+    expect(values.start, equals(0));
+  });
+
+  testWidgets('Range Slider thumbs can be dragged to the min and max (discrete LTR)', (WidgetTester tester) async {
+    RangeValues values = RangeValues(30, 70);
+
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: StatefulBuilder(
+          builder: (BuildContext context, StateSetter setState) {
+            return MediaQuery(
+              data: MediaQueryData.fromWindow(window),
+              child: Material(
+                child: Center(
+                  child: RangeSlider(
+                    values: values,
+                    min: 0,
+                    max: 100,
+                    divisions: 10,
+                    onChanged: (RangeValues newValues) {
+                      setState(() {
+                        values = newValues;
+                      });
+                    },
+                  ),
+                ),
+              ),
+            );
+          },
+        ),
+      ),
+    );
+
+    // Get the bounds of the track by shifting by the the overlay radius.
+    final Offset topLeft = tester.getTopLeft(find.byType(RangeSlider)).translate(24, 0);
+    final Offset bottomRight = tester.getBottomRight(find.byType(RangeSlider)).translate(-24, 0);
+
+    // Drag the start thumb to the min.
+    final Offset leftTarget = topLeft + (bottomRight - topLeft) * 0.3;
+    await tester.dragFrom(leftTarget, topLeft + (bottomRight - topLeft) * -0.4);
+    expect(values.start, equals(0));
+
+    // Drag the end thumb to the max.
+    await tester.pumpAndSettle();
+    final Offset rightTarget = topLeft + (bottomRight - topLeft) * 0.7;
+    await tester.dragFrom(rightTarget, topLeft + (bottomRight - topLeft) * 0.4);
+    expect(values.end, equals(100));
+  });
+
+  testWidgets('Range Slider thumbs can be dragged to the min and max (discrete RTL)', (WidgetTester tester) async {
+    RangeValues values = RangeValues(30, 70);
+
+    await tester.pumpWidget(
+      Directionality(
```

Comment:
I think it may be beneficial to avoid duplication and extract a `buildApp` function

Reply:
This was done for all of the tests that did not need `values`, but there is not much value in doing it for the test that depend on setting the state of `values` (no pun intended)